File: linked_lists/remove_all_duplicates_from_linked_list.rb
def delete_duplicates(head)
    return head if head.nil? || head.next.nil?

    stack = []
    repeats = []
    current = head

    while current

      count = stack.count(current.val)
        if count == 0
            stack << current.val
        else
            repeats << current.val
        end

      current = current.next
    end

    return nil if stack.empty?

    stack - repeats
  end

File: linked_lists/remove_all_duplicates_from_linked_list.py
def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
    if not head or not head.next:
        return head
    
    dummy = ListNode(0)
    dummy.next = head
    prev = dummy
    current = head
    
    while current:
        if current.next and current.val == current.next.val:
            while current.next and current.val == current.next.val:
                current = current.next
            prev.next = current.next
        else:
            prev = prev.next
        current = current.next
    
    return dummy.next